database/src/postgres/tx_indexer.rs

```diff
@@ -0,0 +1,267 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use bigdecimal::num_traits::ToPrimitive;
+use bigdecimal::BigDecimal;
+use sqlx::QueryBuilder;
+
+#[async_trait]
+impl crate::base::tx_indexer::TxIndexerDbManager for crate::postgres::PostgresDBManager {
+    async fn create_tx_tables(&self) -> Result<()> {
+        // Create partitioned transactions table and partitions on each shard
+        for pool in self.shards_pool.values() {
+            // Create the partitioned transactions table
+            sqlx::query(
+                r#"
+                CREATE TABLE IF NOT EXISTS transactions (
```

Comment:
I think we should use the iidch migration here. The code will be more readable. We can create another folder database/src/postgres/migrations/tx_db for migrations related to transactions.